crates/astria-sequencer/src/accounts/state_ext.rs

```diff
@@ -69,8 +74,47 @@ fn nonce_storage_key<T: AddressBytes>(address: T) -> String {
 pub(crate) trait StateReadExt: StateRead + crate::assets::StateReadExt {
     #[instrument(skip_all)]
     async fn get_account_balances(&self, address: Address) -> Result<Vec<AssetBalance>> {
+        let ibc_prefixed_balances = self
+            .get_account_balances_ibc_prefixed(address)
```

Comment:
On the one hand it's great that this method is built on top of the lower level `get_account_balances_ibc_prefixed`. On the other hand we could entirely avoid the allocation if we used the stream provided by cnidarium.

Reply:
I can not use the inner function if you'd prefer. The `get_account_balances()` is only used for outside queries and there isn't much overall performance impact from the allocation (from what I can tell)